### PR TITLE
Email Verification Method

### PR DESCRIPTION
**Pull request by osamamagdy**
## What is the change?
1 - Adding string Status(to check whether he is Activated or not) and confirmation code in the user schema 
2 - Modified the signup function as when a new user is saved successfully in the database, he'll be marked as `Pending` instead of  direct login
3 - Then he will receive a confirmation email with a link that contains a unique encrypted confirmation code using nodemailer and jwt.
4 - The link will send a get request to the specific route with the confirmation code and if the user is already present in the database with that code, his account will be activated
5 - Then we redirect the user to the login page with a message that his account is finally activated.

NOTE : 
The Email and password of the send service must be written in the .env file like this 
`EMAIL = emailaddress@gmail.com`
`PASS = 123456789`

## Related issue?
close: #432 

## How was it tested?
I tried many test cases where the user can enter different emails and they all received different unique confirmation emails.

## Checklist:
- [X] Have you followed the [Contribution Guidelines](https://github.com/ALPHAVIO/BlogSite/blob/master/CONTRIBUTING.md) while contributing.
- [X] Have you checked there aren't other open [Pull Requests](https://github.com/ALPHAVIO/BlogSite/pulls) for the same update/change?
- [] Have you made corresponding changes to the documentation?
- [X] Your submission doesn't break any existing feature.
- [X] Have you tested the code before submission?

### CODE REVIEW

File: models/Blog.model.js [hunk 1]
```diff
@@ -19,11 +19,12 @@ const blogSchema = {
...
+//Making a MongoDB model for the schema-
+module.exports = mongoose.model('Blog', blogSchema);
     likes: {
         type: Number,
         default: 0,
     },
 };
-
-//Making a MongoDB model for the schema-
-module.exports = mongoose.model('Blog', blogSchema);
```
Comment (Pranav016) [inline models/Blog.model.js:29]: why was this moved up ?

**Review by Pranav016 — CHANGES_REQUESTED** [review]
Changes look good to merge, code is formatted 👍 
One small query that I asked.

Also next time make PR from a new branch, that's the best practice. Never commit to master.

File: models/Blog.model.js (hunk 1 above)
Comment (osamamagdy) [reply models/Blog.model.js:29]: I think it was giving me a merge conflict because reformatting but it didn't had any effect

**Comment by osamamagdy** [comment]
I think I accidentally closed this PR
Can you reopen it @Pranav016 @AnanyaNegi @RushanKhan1

**Comment by osamamagdy** [comment]
I also made some changes in a new branch but I didn't understand what happened all of a sudden

**Comment by Pranav016** [comment]
Not getting option to reopen PR, I think the branch was deleted..

**Comment by osamamagdy** [comment]
I created another PR at #631 with the same changes. please review @Pranav016 @AnanyaNegi @RushanKhan1

**Pull request closed without merging**